service/src/main/java/uk/gov/hmcts/reform/fpl/service/ApplicationDocumentsService.java

```diff
@@ -0,0 +1,86 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.model.ApplicationDocument;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+import uk.gov.hmcts.reform.fpl.utils.DocumentUploadHelper;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static io.jsonwebtoken.lang.Collections.isEmpty;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.findElement;
+
+@Service
+@RequiredArgsConstructor(onConstructor_ = {@Autowired})
+public class ApplicationDocumentsService {
+
+    private final Time time;
+    private final DocumentUploadHelper documentUploadHelper;
+
+    public Map<String, Object> updateApplicationDocuments(List<Element<ApplicationDocument>> currentDocuments,
```

Comment:
Why return a map, would it not be nicer to just return the list of updated documents?

Reply:
leaving this as is for now, as late in story cycle as mentioned